sequencer/dbmanager.go

```diff
@@ -200,9 +202,59 @@ func (d *dbManager) StoreProcessedTxAndDeleteFromPool(ctx context.Context, tx tr
 		}
 	}
 
+	// Send tx data to data stream server
+	if d.streamServer != nil {
+		l2BlochHeader, err := d.GetLastL2BlockHeader(ctx, dbTx)
+		if err != nil {
+			return err
+		}
+
+		l2Block := DSL2Block{
+			BatchNumber:    tx.batchNumber,
+			L2BlockNumber:  l2BlochHeader.Number.Uint64(),
+			Timestamp:      tx.timestamp,
+			GlobalExitRoot: batch.GlobalExitRoot,
+			Coinbase:       tx.coinbase,
+		}
+
+		l2Transaction := DSL2Transaction{
+			BatchNumber:                 batch.BatchNumber,
+			EffectiveGasPricePercentage: uint8(tx.response.EffectivePercentage),
+			IsValid:                     1,
+			EncodedLength:               uint32(len(txData)),
+			Encoded:                     txData,
+		}
+
+		err = d.streamServer.StartAtomicOp()
+		if err != nil {
+			return err
+		}
+
+		_, err = d.streamServer.AddStreamEntry(EntryTypeL2Block, l2Block.Encode())
+		if err != nil {
+			return err
+		}
+
+		_, err = d.streamServer.AddStreamEntry(EntryTypeL2Tx, l2Transaction.Encode())
+		if err != nil {
+			return err
+		}
+	}
+
 	err = dbTx.Commit(ctx)
 	if err != nil {
+		err2 := d.streamServer.RollbackAtomicOp()
```

Comment:
In line 244 I see `err = dbTx.Commit(ctx)` but I don't see where we do the `dbTx.Rollback` in case we get an error